### PR TITLE
Changed 'index-separator' flag to 'no-index-separator' flag

### PR DESCRIPTION
**Pull request by jenniferliddle**

### CODE REVIEW

**Comment by dkj** [comment]
Is this tested? Didn't work on a real data for me. We need realistic tests dependent on this change i.e. some test data with indexed reads.

**Comment by dkj** [comment]
To avoid the semantic change, it may be "nicer" to drop the `-s` and have a `-S` for the short variant of the new `--no-index-separator` option.

File: src/i2b.c [hunk 1]
```diff
@@ -479,8 +480,8 @@ static opts_t* i2b_parse_args(int argc, char *argv[])
     if (!opts->library_name) opts->library_name = strdup("unknown");
     if (!opts->sample_alias) opts->sample_alias = strdup(opts->library_name);
     if (!opts->sequencing_centre) opts->sequencing_centre = strdup("SC");
-    if (va_isEmpty(opts->barcode_tag)) va_push(opts->barcode_tag,strdup(DEFAULT_BARCODE_TAG));
-    if (va_isEmpty(opts->quality_tag)) va_push(opts->quality_tag,strdup(DEFAULT_QUALITY_TAG));
+    if (va_isEmpty(opts->barcode_tag)) while (opts->barcode_tag->end < 10) va_push(opts->barcode_tag,strdup(DEFAULT_BARCODE_TAG));
```
Comment (dkj) [inline src/i2b.c]: Not sure I understand this: would a `MAX_DEFAULT_BARCODES` define describe it appropriately?

Comment (jenniferliddle) [reply src/i2b.c]: Yes, it would be clearer with a #define.

**Comment by dkj** [comment]
We need other changes to Bambi for separated indexes in production.

**Pull request merged**